server/plugins/task-plugin-mysql/src/main/java/com/oceanbase/odc/plugin/task/mysql/datatransfer/job/datax/DataXTransferJob.java

```diff
@@ -0,0 +1,211 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.alibaba.datax.common.element.ColumnCast;
+import com.alibaba.datax.common.util.Configuration;
+import com.alibaba.datax.common.util.MessageSource;
+import com.alibaba.datax.core.job.JobContainer;
+import com.alibaba.datax.core.statistics.communication.Communication;
+import com.alibaba.datax.core.statistics.communication.CommunicationTool;
+import com.alibaba.datax.core.statistics.container.communicator.job.StandAloneJobContainerCommunicator;
+import com.alibaba.datax.core.util.container.CoreConstant;
+import com.alibaba.datax.core.util.container.LoadUtil;
+import com.oceanbase.odc.common.util.StringUtils;
+import com.oceanbase.odc.plugin.task.api.datatransfer.model.ObjectResult;
+import com.oceanbase.odc.plugin.task.mysql.datatransfer.common.Constants;
+import com.oceanbase.odc.plugin.task.mysql.datatransfer.job.AbstractJob;
+import com.oceanbase.odc.plugin.task.mysql.datatransfer.job.datax.model.JobConfiguration;
+import com.oceanbase.odc.plugin.task.mysql.datatransfer.job.datax.model.parameter.TxtWriterPluginParameter;
+import com.oceanbase.odc.plugin.task.mysql.datatransfer.job.datax.util.DataXJobIdProvider;
+import com.oceanbase.tools.loaddump.common.model.ObjectStatus.Status;
+
+public class DataXTransferJob extends AbstractJob {
+    private static final Logger LOGGER = LoggerFactory.getLogger("DataTransferLogger");
+    private static final Pattern DATA_FILE_PATTERN =
+            Pattern.compile("(^\"?(.+)\"?.(sql|csv|dat|txt))__(.+)$", Pattern.CASE_INSENSITIVE);
+
+    private final JobConfiguration jobConfig;
+    /**
+     * for monitoring, {@link StandAloneJobContainerCommunicator} need this to collect. Each job should
+     * own a unique ID.
+     */
+    private final Long jobId;
+    private JobContainer jobContainer;
+    private StandAloneJobContainerCommunicator containerCommunicator;
+
+    public DataXTransferJob(ObjectResult object, JobConfiguration jobConfig) {
+        super(object);
+        this.jobConfig = jobConfig;
+        this.jobId = DataXJobIdProvider.getInstance().fetch();
+    }
+
+    @Override
+    public void run() throws Exception {
+
+        DataXJobMonitor monitor = new DataXJobMonitor();
+
+        try {
+            Configuration configuration = ConfigurationResolver.resolve(jobConfig);
+            configuration.set(CoreConstant.DATAX_CORE_CONTAINER_JOB_ID, jobId);
+            this.containerCommunicator = new StandAloneJobContainerCommunicator(configuration);
+
+            new Thread(monitor, "DataX-Monitor-" + Thread.currentThread().getName()).start();
```

Comment:
how to stop it? prefer not to new a thread directly~

Reply:
This monitor will be stopped in finally block by calling it's stop method. In the `stop`, a stop flag will be set into true to break loop.

```JAVA
        DataXJobMonitor monitor = new DataXJobMonitor();
        try {
           ...
            new Thread(monitor, "DataX-Monitor-" + Thread.currentThread().getName()).start();
           ...
        } finally {
            monitor.stop();
        }

```